src/net/ossl.cc

```diff
@@ -838,9 +908,19 @@ class session : public enable_shared_from_this<session>, public session_impl {
                 throw ossl_error("Failed to load cert/key pair");
             }
         }
+        SSL_CTX_set1_cert_store(_ctx.get(), *_creds);
         return ssl_ctx;
     }
 
+    static std::optional<sstring> get_ossl_string(X509_NAME* name){
+        if (auto name_str = X509_NAME_oneline(name, nullptr, 0)) {
+            sstring ossl_str(name_str);
+            OPENSSL_free(name_str);
```

Comment:
sstring constructor allocates and can therefore throw. Worth wrapping in a unique_ptr with custom deleter.